CONTRIBUTING.md

```diff
@@ -0,0 +1,278 @@
+   ```
+   must be placed before each struct, and then those of these traits, which can't
+   be auto-derived because of the member field restrictions should be removed.
+
+2. `Default` derivation should be performed whenever there is a rationale to 
+   have default constructor initializing "empty" data structure, i.e. this 
+   empty structure has a real use in the business logic *outside of the scope
+   of testing or creating dumb data*. For instance, if the structure consists
+   only of collection types which may be empty it should derive `Default` trait.
+
+3. **Error types** (both structs and enums) must implement `Display` and `Error`
+   traits manually, and should provide `Error::source` function if some of the
+   error cases contain other error type.
+
+4. `Display` should be implemented for all data types which may be presented to
+   the end user (not developers!), for instance in command line or as a part of
+   GUI. Here are some guidelines: 
+   - Normally, `Display` implementation should not just repeat `Debug` and
+     structure the data in some visually-acceptable way. 
+   - One should pay attention to the ability of providing alternative ways of 
+     data formatting with `{:#}` formatting string option, detectable by 
+     `std::fmt::Formatter::alternate()` function. Other important options to
+     look at are `align`, `fill`, `pad`, `precision` and `width`. 
+   - When displaying the member fields it is important to consider the ability
+     to pass them display formatting options; thus, 
+     `Display::fmt(&self.field, f)?;` is preferable over 
+     `write!(f, "{}", self.field)?;`
+
+5. Serde serializers should be implemented for all data types which may persist
+   or may be presented in the UI or API as JSON/YAML and other kinds of data 
+   representations (in fact, these are all data types).
+
+The discussion about trait derivation can be read at 
+[the tracking issue](https://github.com/rust-bitcoin/rust-bitcoin/issues/555).
+
+### MSRV
+
+The Minimal Supported Rust Version (MSRV) is 0.29; it is enforced by our CI. 
+Later we plan to increase MSRV to support Rust 2018 and you are welcome to check
+the [tracking issue](https://github.com/rust-bitcoin/rust-bitcoin/issues/510).
+
+### Naming conventions
+
+Naming of data structures/enums and their fields/variants must follow names used
+in Bitcoin Core, with except to:
+- case, which should follow Rust standards (i.e. PascalCase for types and 
+  snake_case for fields and variants)
+- `C`-prefix, which should be omitted
+
+
+## Security
+
+Security is the primary focus of Rust-LNPBP; disclosure of security
+vulnerabilities helps prevent user loss of funds. If you believe a vulnerability
+may affect other  implementations, please inform them.
+
+Note that Rust-LNPBP is currently considered "pre-production" during this time,
+there is no special handling of security issues. Please simply open an issue on
+Github.
```

Comment:
Is it? At least `electrs` depends on it and I consider it in-production.

Reply:
This appears to have been copied from LNPBP and isn't right.